rust_src/src/editfns.rs

```diff
@@ -152,3 +152,27 @@ pub fn goto_char(position: LispObject) -> LispObject {
     };
     position
 }
+
+/// Return the byte position for character position POSITION.
+/// If POSITION is out of range, the value is nil.
+#[lisp_fn]
+pub fn position_bytes(position: LispObject) -> LispObject {
+    if let LispNumber::Fixnum(pos) = position.as_number_coerce_marker_or_error() {
+        let cur_buf = ThreadState::current_buffer();
+
+        // cast up to 64bit before comparing with pos
+        let begv = cur_buf.begv as EmacsInt;
+        let zv = cur_buf.zv as EmacsInt;
+
+        if pos < begv || pos > zv {
+            return LispObject::constant_nil();
+        }
+
```

Comment:
I think you don't need the cast here. This can be done in one if else block with implicit returning. Take a look how the *or_error methods are used in other functions. That's what I do when I'm not sure how to use a function :)